sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveAliasesSuite.scala

```diff
@@ -88,4 +94,46 @@ class ResolveAliasesSuite extends AnalysisTest {
     checkAliasName(t1.select(DateSub(Literal(Date.valueOf("2021-01-18")), Literal(null))),
       "date_sub(DATE '2021-01-18', NULL)")
   }
+
+  test("SPARK-40822: Stable derived column aliases") {
+    withSQLConf(SQLConf.STABLE_DERIVED_COLUMN_ALIAS_ENABLED.key -> "true") {
+      Seq(
+        // Literals
+        "' 1'" -> "' 1'",
+        """"abc"""" -> """"abc"""",
```

Comment:
It makes me think if we should always use the standard SQL syntax to generate the alias name, instead of using the original SQL string. The string literal accepts two quote symbols and seems better to normalize it when generating alias names. cc @srielau

Reply:
You mean convert " to '  for String literals? +1

Reply:
How should I distinguish double quoted string literals from double quoted identifiers on this level?

Reply:
I have added such conversion.